File: Suin/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = 'Yes'
    cards1_idx, cards2_idx = 0, 0

    for word in goal:
        #조건문은 앞에서부터 차례대로 판단됨
        if cards1_idx<len(cards1) and word == cards1[cards1_idx]:
            cards1_idx +=1
        elif cards2_idx<len(cards2) and word == cards2[cards2_idx]:
            cards2_idx +=1
        else: 
            answer = 'No'
            break

    return answer

File: JoonSeong/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int idx1 = 0, idx2 = 0;
        boolean flag = true;
        for(int i = 0; i < goal.length; i++) {
            if(cards1[idx1].equals(goal[i])) {
                if(idx1 + 1 < cards1.length) idx1++;
            } else if(cards2[idx2].equals(goal[i])) {
                if(idx2 + 1 < cards2.length) idx2++;
            } else {
                flag = false;
                break;
            }
        }
        if(flag) {
            return "Yes";
        } else {
            return "No";
        }
    }
}